packages/mui-base/src/useSwitch/useSwitch.types.ts

```diff
@@ -49,3 +49,27 @@ interface UseSwitchInputSlotOwnProps {
 
 export type UseSwitchInputSlotProps<TOther = {}> = Omit<TOther, keyof UseSwitchInputSlotOwnProps> &
   UseSwitchInputSlotOwnProps;
+
+export interface UseSwitchReturnValue {
+  /**
+   * If `true`, the component will be checked.
+   */
+  checked: boolean;
+  /**
+   * If `true`, the component will be disabled.
+   */
+  disabled: boolean;
+  /**
+   * If `true`, it indicates that the component is being focused using keyboard.
+   * @default false
+   */
+  focusVisible: boolean;
+  /**
+   * Function to get the props to place on an HTML `input` to make the switch accessible.
```

Comment:
The following change https://github.com/mui/material-ui/pull/36050#discussion_r1099974860 was not applied.